components/Search/index.js

```diff
@@ -0,0 +1,284 @@
+import React, { Component } from 'react'
+import { css } from 'glamor'
+import debounce from 'lodash.debounce'
+
+import { Router } from '../../lib/routes'
+
+import { DEFAULT_FILTERS } from './constants'
+import {
+  deserializeFilters,
+  serializeFilters,
+  deserializeSort,
+  serializeSort
+} from './serialize'
+
+import Input from './Input'
+import Results from './Results'
+
+import {
+  Center,
+  mediaQueries
+} from '@project-r/styleguide'
+
+const styles = {
+  container: css({
+    padding: '15px 15px 120px',
+    [mediaQueries.mUp]: {
+      padding: '40px 0 120px'
+    }
+  })
+}
+
+class Search extends Component {
+  constructor (props, ...args) {
+    super(props, ...args)
+
+    this.state = {
+      loading: false,
+      loadingFilters: false,
+      searchQuery: '',
+      filterQuery: '',
+      submittedQuery: '',
+      filters: DEFAULT_FILTERS,
+      serializedFilters: '',
+      sort: {
+        key: 'publishedAt'
+      },
+      serializedSort: '',
+      totalCount: 0,
+      isMobile: true,
+      allowFocus: true
+    }
+
+    this.loadFilters = debounce(() => {
+      this.setState({
+        filters:
+          this.state.filterQuery !== this.state.searchQuery
+            ? DEFAULT_FILTERS
+            : this.state.filters,
+        filterQuery: this.state.searchQuery,
+        loadingFilters: false
+      })
+    }, 200)
+
+    this.onInputChange = (_, value) => {
+      this.setState({
+        searchQuery: value,
+        loadingFilters: true,
+        allowFocus: true
+      })
+      this.loadFilters()
+    }
+
+    this.onSearch = () => {
+      const sort = {
+        key: 'relevance'
+      }
+      this.setState({
+        submittedQuery: this.state.searchQuery,
+        filterQuery: this.state.searchQuery,
+        filters: DEFAULT_FILTERS,
+        sort,
+        allowFocus: !this.state.isMobile
+      })
+      this.updateUrl(undefined, serializeSort(sort))
+      window._paq.push(['trackSiteSearch',
+        this.state.searchQuery,
+        false,
```

Comment:
maybe add filters here as category? Otherwise seeing 0 results for something might not make sense without knowing the filters.

Reply:
Done. We're now also tracking the filters along with the count on filter click, while this one remains as is (filters are reset on search).